server/src/main/java/org/opensearch/index/store/remote/utils/TransferManager.java

```diff
@@ -95,6 +96,14 @@ public IndexInput fetchBlob(BlobFetchRequest blobFetchRequest) throws IOExceptio
     @SuppressWarnings("removal")
     private static FileCachedIndexInput createIndexInput(FileCache fileCache, StreamReader streamReader, BlobFetchRequest request) {
         try {
+            // This local file cache is ref counted and may not strictly enforce configured capacity.
+            // If we find available capacity is exceeded, deny further BlobFetchRequests.
+            Supplier<Short> cacheUsagePerc = () -> fileCache.capacity() <= 0
```

Comment:
No need for a supplier followed by a `.get()` call.

Also I don't think we need to be working with percentages here because we can just compare byte values, i.e. if usage + to be requested bytes > capacity then fail the request.

Reply:
Agreed, that was more about the  preemptive cleaning 👍

Reply:
Would it make sense for this failure threshold to be a configurable cluster setting?

Reply:
I'm not sure how it would be distinct from the already configurable cache capacity. Any threshold you configure would essentially be the 'real' capacity.